### PR TITLE
feat: Validation usability improvements

### PR DESCRIPTION
**Pull request by antroy-madetech**
Validation catches all errors now rather than it being a manual process to iterate over failures
Undeprecated the make-addons and pipeline generate commands, as they are used in different places in the new process.

### CODE REVIEW

File: dbt_platform_helper/commands/pipeline.py [hunk 1]
```diff
@@ -28,13 +28,9 @@ def pipeline():
     check_platform_helper_version_needs_update()
 
 
-@pipeline.command(deprecated=True, hidden=True)
+@pipeline.command()
```
Comment (gabelton) [inline dbt_platform_helper/commands/pipeline.py:31]: What's the scenario where this gets run by itself?

**Comment by gabelton** [comment]
Is it possible to update / add a unit test to capture the change to validation capturing more than one error?

**Comment by antroy-madetech** [comment]
> Is it possible to update / add a unit test to capture the change to validation capturing more than one error?

Thought I had, but it was just multiple bad environments I'd added. Adding a separate test now.

**Review by WillGibson — APPROVED** [review]
Lovely

File: dbt_platform_helper/commands/pipeline.py (hunk 1 above)
Comment (gabelton) [reply dbt_platform_helper/commands/pipeline.py:31]: makes sense. thanks!

**Review by gabelton — APPROVED** [review]

**Pull request merged**